File: 프로그래머스/lv3/43162. 네트워크/네트워크.js
function solution(n, computers) {
    let count = 0;
    const visited = Array.from({length: n}, () => false);
    
    const dfs = (node) => {
        visited[node] = true;
        const connectionArr = computers[node];

        for (let i = 0; i < connectionArr.length; i++) {
            if (connectionArr[i] === 1 && !visited[i]) {
                dfs(i);
            }
        }
    }
    
    for (let node = 0; node < computers.length; node++) {
        if (!visited[node]) { 
            dfs(node);
            count++;
        }
    }
    
    return count;
}

File: 프로그래머스/3/43162. 네트워크/네트워크.py
def solution(n, computers):
    answer = 0
    
    graph = []
    
    for i, connections in enumerate(computers):
        neighbors = []
        for j, connected in enumerate(connections):
            if i != j and connected == 1:
                 neighbors.append(j)
        graph.append(neighbors)
        
    
    visited = [False] * n
    
    def bfs(start): # 시작 지점부터 연결된 네트워크 모두 탐색하여 방문 처리
        queue = [start]
        visited[start] = True
        
        while queue:
            cur = queue.pop(0) 
            for neighbor in graph[cur]:
                if not visited[neighbor]:
                    queue.append(neighbor)
                    visited[neighbor] = True
            
        
    # graph: [[1], [0], []]
    for i in range(len(graph)):
        if not visited[i]: # 새로운 네트워크 발견
            bfs(i)
            answer += 1
    
    return answer